sled-agent/src/bootstrap/agent.rs

```diff
@@ -130,13 +128,31 @@ impl Agent {
         sled_config: SledConfig,
         address: Ipv6Addr,
     ) -> Result<Self, BootstrapError> {
+        let data_link = if let Some(link) = sled_config.data_link.clone() {
+            link
+        } else {
+            Dladm::find_physical().map_err(|err| {
+                BootstrapError::SledError(format!(
+                    "Can't access physical link, and none in config: {}",
+                    err
+                ))
+            })?
+        };
```

Comment:
Admittedly, this was one other thing I did in this PR - there is a code snippet to "access an `Option<PhysicalLink>`, and use `dladm` to look up a default link if one hasn't been supplied".

I moved these calls up in the call-stack, which simplifies the amount of code that needs to handle "dladm" errors.